R/expectation.r

```diff
@@ -11,20 +11,73 @@
 #' @keywords internal
 #' @export
 expectation <- function(passed, failure_msg, success_msg = "unknown", srcref = NULL) {
-  structure(
+  new_expectation(passed = passed, failure_msg = failure_msg,
+                  success_msg = success_msg, srcref = srcref)
+}
+
+new_expectation <- function(failure_msg, success_msg, srcref, ...,
+                            passed = FALSE, error = FALSE, skipped = FALSE) {
+  if (passed) {
```

Comment:
If `error`, then `passed` is `FALSE` by construction (although not enforced). I agree that a type is better, but I'd like to deal with this in a new batch of changes, and also change the reporters -- this PR is already big enough.